src/libraries/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.Date.cs

```diff
@@ -42,103 +42,73 @@ public static void WriteDateTimeOffsetTrimmed(Span<byte> buffer, DateTimeOffset
-
-                // Remove trailing zeros
-                while (true)
-                {
-                    (uint quotient, uint remainder) = DivRem(fraction, 10);
-                    if (remainder != 0)
-                    {
-                        break;
-                    }
-                    fraction = quotient;
-                    numFractionDigits--;
-                }
-
-                // The last fraction digit's index will be (the period's index plus one) + (the number of fraction digits minus one)
-                int fractionEnd = 19 + numFractionDigits;
-
-                // Write fraction
-                // Leading zeros are written because the fraction becomes zero when it's their turn
-                for (int i = fractionEnd; i > curIndex; i--)
-                {
-                    buffer[i] = (byte)((fraction % 10) + (uint)'0');
-                    fraction /= 10;
-                }
-
-                curIndex = fractionEnd + 1;
+                // When there is nothing to trim we are done.
+                bytesWritten = buffer.Length;
+                return;
             }
 
-            bytesWritten = curIndex;
-
             // We are either trimming a DateTimeOffset, or a DateTime with
             // DateTimeKind.Local or DateTimeKind.Utc
-            if (buffer.Length > JsonConstants.MaximumFormatDateTimeLength)
+            if (buffer.Length == maxLenNoOffset)
+            {
+                bytesWritten = curIndex;
+            }
+            else
             {
                 // Write offset
 
-                buffer[curIndex] = buffer[27];
-
-                // curIndex is at one of 'Z', '+', or '-'
-                bytesWritten = curIndex + 1;
-
-                // We have a Non-UTC offset i.e. (+|-)hh:mm
-                if (buffer.Length == JsonConstants.MaximumFormatDateTimeOffsetLength)
+                if (buffer.Length == maxLenWithOffset)
                 {
-                    // Last index of the offset
-                    int bufferEnd = curIndex + 5;
-
-                    // Cache offset characters to prevent them from being overwritten
-                    // The second minute digit is never at risk
-                    byte offsetMinDigit1 = buffer[31];
-                    byte offsetHourDigit2 = buffer[29];
-                    byte offsetHourDigit1 = buffer[28];
-
-                    Debug.Assert(buffer[30] == JsonConstants.Colon);
-
-                    // Write offset characters
-                    buffer[bufferEnd] = buffer[32];
-                    buffer[bufferEnd - 1] = offsetMinDigit1;
-                    buffer[bufferEnd - 2] = JsonConstants.Colon;
-                    buffer[bufferEnd - 3] = offsetHourDigit2;
-                    buffer[bufferEnd - 4] = offsetHourDigit1;
+                    // We have a Non-UTC offset i.e. (+|-)hh:mm
+
+                    // Write offset characters left to right to prevent overwriting owerselfs
```

Comment:
Presumably you meant
```suggestion
                    // Write offset characters left to right to prevent overwriting themselves
```

Reply:
maibee ;)